packages/react-link/src/components/Link/Link.types.ts

```diff
@@ -1,73 +1,78 @@
 import * as React from 'react';
-import type { ComponentPropsCompat } from '@fluentui/react-utilities';
+import type { ComponentProps, ComponentState, ObjectShorthandProps } from '@fluentui/react-utilities';
 
-/**
- * {@docCategory Link}
- */
-export type LinkProps = ComponentPropsCompat &
-  React.AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement> &
-  Omit<React.ButtonHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement>, 'type'> & {
-    /**
-     * URL the link points to. If not provided, the link renders as a button (unless that behavior is
-     * overridden using `as`).
-     */
-    href?: string;
+export type LinkSlots = {
+  root:
+    | ObjectShorthandProps<JSX.IntrinsicElements['a'], HTMLAnchorElement, 'a'>
+    | ObjectShorthandProps<JSX.IntrinsicElements['button'], HTMLButtonElement, 'button'>;
```

Comment:
FYI with my PR #19642 that I just merged, this will need to be updated to:
```suggestion
  root: IntrinsicShorthandProps<'a', 'button'>;
```

Reply:
Actually, I think this might be an exception to how the new IntrinsicShorthandProps works: in addition to "as", it also uses "href" to distinguish between `<a>` and `<button>`. 

I tried to figure out how to make that work in the type system, but I don't think TypeScript can handle using two different discriminant properties (either `a` or `href`). Ultimately, I think your best bet would be to treat both `a` and `button` as default props (so it allows the union of all `a` and `button` props, unless "as" is provided).
```suggestion
  root: IntrinsicShorthandProps<'a'> | IntrinsicShorthandProps<'button'>;
```

Reply:
This has been fixed now, marking as resolved.